connector/connect/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -62,3 +65,29 @@ message CreateScalarFunction {
     FUNCTION_LANGUAGE_SCALA = 3;
   }
 }
+
+// As writes are not directly handled during analysis and planning, they are modeled as commands.
```

Comment:
We want this to be eagerly executed, so - IMO - making this a command is your only option.